src/LightningDB/LightningCursor.cs

```diff
@@ -227,17 +230,38 @@ private bool Get(CursorOperation operation)
         }
 
         private bool Get(CursorOperation operation, byte[] key)
+        {
+            if (key is null)
+                throw new ArgumentNullException(nameof(key));
+
+            return Get(operation, key.AsSpan());
+        }
+
+        private unsafe bool Get(CursorOperation operation, ReadOnlySpan<byte> key)
         {
             _currentValue = default;
-            var findKey = new MDBValue(new Span<byte>(key));
-            return mdb_cursor_get(_handle, ref findKey, ref _currentValue, operation) == 0;
+#warning should this update _currentKey?
```

Comment:
Yes, it looks like I introduced this in my first pass at changing things. Good catch.

Reply:
Question For these Gets, if the key doesn't exist, should the cursor data be updated?
The LMDB docs don't actually say what might get returned if the key isn't found

Reply:
I believe it's using search under the covers and if my memory is correct it will change the cursor index to the one just below a key "higher" than what you looked for if that makes sense.

Reply:
That makes sense to me. One last update one the way then